tests/integ/sagemaker/experiments/test_run.py

```diff
@@ -134,11 +134,15 @@ def test_run_name_vs_trial_component_name_edge_cases(sagemaker_session, input_na
         ) as run1:
             assert not run1._experiment.tags
             assert not run1._trial.tags
-            is_run_tc = is_run_trial_component(
-                trial_component_name=run1._trial_component.trial_component_name,
-                sagemaker_session=sagemaker_session,
-            )
-            assert is_run_tc
+
+            def verify_is_run():
+                is_run_tc = is_run_trial_component(
+                    trial_component_name=run1._trial_component.trial_component_name,
+                    sagemaker_session=sagemaker_session,
+                )
+                assert is_run_tc
+
+            retry_with_backoff(verify_is_run, 4)
```

Comment:
After a second look, there're retries already in the helper method https://github.com/aws/sagemaker-python-sdk/blob/162e922747a66c972faf6dcbdd2b779f6400170a/src/sagemaker/experiments/_utils.py#L204.

I guess the unstable issue in pre-prod is relating to the delays on tag update for search operation? 
Wondering 
- why the behavior differentiates in prod and preprod. 
- And why the SDK integ tests are running in preprod env? I thought it's running in beta and prod only

Reply:
* the failure that motiviates this change was in beta
* not sure where you are referring to difference in behavior between prod and preprod - can you specify where?